mkFit/MkBuilder.cc

```diff
@@ -1300,20 +1300,22 @@ void MkBuilder::find_duplicates(TrackVec& tracks)
       {
 	if(Config::useHitsForDuplicates)
 	{
-	  std::vector<int> vecOfHits;
 	  float numHitsShared = 0;
-	  for (int ihit = 0; ihit < track.nTotalHits(); ihit++)
-	  {
-	    vecOfHits.push_back(track.getHitIdx(ihit));
-	  }
+	  const std::vector<HitOnTrack> *hots = track.getHitsOnTrackVector();
 	  for (int ihit2 = 0; ihit2 < track2.nTotalHits(); ihit2++)
-	  {
-	    std::vector<int>::iterator it;
-	    it = std::find(vecOfHits.begin(), vecOfHits.end(),track2.getHitIdx(ihit2) );
-	    if (it != vecOfHits.end()) numHitsShared++;
-	  }
-	  float fracHitsShared = (track.nTotalHits() < track2.nTotalHits()) ? numHitsShared/track.nTotalHits() : numHitsShared/track2.nTotalHits();
-	  //Only remove one of the tracks if they share at least 90% of the hits (denominator is the shorter track)
+	    {
+	      int hitidx2 = track2.getHitIdx(ihit2);
+	      int hitlyr2 = track2.getHitLyr(ihit2);
+	      if(hitidx2 >=0)
+	      {
+		auto it = std::find_if (hots->begin(), hots->end(),[&hitidx2,&hitlyr2](const HitOnTrack& element){
```

Comment:
I wonder what would be the difference wrt. doing
```cpp
std::find_if (track.BeginHitsOnTrack(), track.EndHitsOnTrack(), ...
```
with the current interface? (i.e. the relation of `track.hitsOnTrk_.size()` to `track.lastHitIdx_ + 1`)

Reply:
Thanks for the comment! I tested your suggestion, and it gives the exact same performance, without the need for adding a new function to Track.h. The latest commit includes these changes.